### DISCUSSION TITLE
Client side image compression

### DISCUSSION THREAD

**Discussion by KasperDjernaes** [sveltejs/kit#8840] [Q&A]
I am trying to perform client side image compression before it's being send to the server.

I'm thinking I should be able to do this with a custom enhance method or event listenener.
However, my main problem is that I do not know how I would alter the form data before it is send to the server? even if I managed to get the file resized on the client, I'm not sure how to send it afterwards.

I'm using a form element like this to do the upload:

`<input type="file" id="stuffImage" name="stuffImage" accept="image/png, image/jpeg" >`

**Comment by NormandoHall** [comment]
I am using [Filepond](https://github.com/pqina/filepond). It is superb and highly customizable. It has [svelte wrappe](https://github.com/pqina/svelte-filepond)r also. Enjoy 🍺
